lib/elixir/test/elixir/module/types/integration_test.exs

```diff
@@ -688,6 +688,75 @@ defmodule Module.Types.IntegrationTest do
 
       assert_warnings(files, warning)
     end
+
+    test "collapses list types and fails to unify" do
```

Comment:
We should generally avoid integration tests because they become quite expensive over time. Ideally we would reduce this test case to a minimum necessary. You can check `types_test.exs` in the same directory for more unit like test. You should be able to reproduce this with:

```elixir
        warning(
          [map1, map2],
          (
            [_var1, _var2] = [map1, map2]
            %{} = map1
            %{} = map2.subkey
          )
        )
```